File: src/Entity/Event.php
<?php

namespace App\Entity;

use App\Repository\EventRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\Exception\FileException;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Symfony\Component\String\Slugger\SluggerInterface;

/**
 * @ORM\Entity(repositoryClass=EventRepository::class)
 */
class Event
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $nom;

    /**
     * @ORM\Column(type="integer")
     */
    private $nbr_place;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $depart;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $arivee;

    /**
     * @ORM\Column(type="datetime")
     */
    private $date_allee;

    /**
     * @ORM\Column(type="datetime")
     */
    private $date_retour;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $description;

    /**
     * @ORM\ManyToMany(targetEntity=Participant::class, mappedBy="id_event")
     */
    private $inscription;

    public function __construct()
    {
        $this->inscription = new ArrayCollection();
    }


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getNbrPlace(): ?int
    {
        return $this->nbr_place;
    }

    public function setNbrPlace(int $nbr_place): self
    {
        $this->nbr_place = $nbr_place;

        return $this;
    }

    public function getDepart(): ?string
    {
        return $this->depart;
    }

    public function setDepart(string $depart): self
    {
        $this->depart = $depart;

        return $this;
    }

    public function getArivee(): ?string
    {
        return $this->arivee;
    }

    public function setArivee(string $arivee): self
    {
        $this->arivee = $arivee;

        return $this;
    }

    public function getDateAllee(): ?\DateTimeInterface
    {
        return $this->date_allee;
    }

    public function setDateAllee(\DateTimeInterface $date_allee): self
    {
        $this->date_allee = $date_allee;

        return $this;
    }

    public function getDateRetour(): ?\DateTimeInterface
    {
        return $this->date_retour;
    }

    public function setDateRetour(\DateTimeInterface $date_retour): self
    {
        $this->date_retour = $date_retour;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    /**
     * @return Collection|Participant[]
     */
    public function getInscription(): Collection
    {
        return $this->inscription;
    }

    public function addInscription(Participant $inscription): self
    {
        if (!$this->inscription->contains($inscription)) {
            $this->inscription[] = $inscription;
            $inscription->addIdEvent($this);
        }

        return $this;
    }

    public function removeInscription(Participant $inscription): self
    {
        if ($this->inscription->removeElement($inscription)) {
            $inscription->removeIdEvent($this);
        }

        return $this;
    }


}

File: E-bike_java/src/entities/Event.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;
import java.util.Date;
/**
 *
 * @author amine
 */
public class Event {
    private int id;
    private String nom;
    private int nbr_place;
    private String depart;
    private String arivee;
    private  Date  date_allee;
    private Date date_retour;
    private String description;
    private String image;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getNbr_place() {
        return nbr_place;
    }

    public void setNbr_place(int nbr_place) {
        this.nbr_place = nbr_place;
    }

    public String getDepart() {
        return depart;
    }

    public void setDepart(String depart) {
        this.depart = depart;
    }

    public String getArivee() {
        return arivee;
    }

    public void setArivee(String arivee) {
        this.arivee = arivee;
    }

    public Date getDate_allee() {
        return date_allee;
    }

    public void setDate_allee(Date date_allee) {
        this.date_allee = date_allee;
    }

    public Date getDate_retour() {
        return date_retour;
    }

    public void setDate_retour(Date date_retour) {
        this.date_retour = date_retour;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Event(int id, String nom, int nbr_place, String depart, String arivee, Date date_allee, Date date_retour, String description, String image) {
        this.id = id;
        this.nom = nom;
        this.nbr_place = nbr_place;
        this.depart = depart;
        this.arivee = arivee;
        this.date_allee = date_allee;
        this.date_retour = date_retour;
        this.description = description;
        this.image = image;
    }

    

    public Event(String nom, int nbr_place, String depart, String arivee, Date date_allee, Date date_retour, String description, String image) {
        this.nom = nom;
        this.nbr_place = nbr_place;
        this.depart = depart;
        this.arivee = arivee;
        this.date_allee = date_allee;
        this.date_retour = date_retour;
        this.description = description;
        this.image = image;
    }

    public Event(int nbr_place, String depart, String arivee, Date date_allee, String description, String image) {
        this.nbr_place = nbr_place;
        this.depart = depart;
        this.arivee = arivee;
        this.date_allee = date_allee;
        this.description = description;
        this.image = image;
    }
    
    

    public Event() {
    }

    @Override
    public String toString() {
        return "Event{ nom=" + nom + ", nbr_place=" + nbr_place + ", depart=" + depart + ", arivee=" + arivee + ", date_allee=" + date_allee + ", date_retour=" + date_retour + ", description=" + description + ", image=" + image + '}';
    }

    public Event(int id, String nom, int nbr_place, String depart, String arivee, Date date_allee, Date date_retour, String description) {
        this.id = id;
        this.nom = nom;
        this.nbr_place = nbr_place;
        this.depart = depart;
        this.arivee = arivee;
        this.date_allee = date_allee;
        this.date_retour = date_retour;
        this.description = description;
    }

  
    
    
    
    
  
    
    
    
    
}
